src/cargo/util/toml/mod.rs

```diff
@@ -1065,11 +1008,119 @@ impl<T> MaybeWorkspace<T> {
     }
 }
 
+type MaybeWorkspaceDependency = MaybeWorkspace<TomlDependency, TomlWorkspaceDependency>;
+
+#[derive(Deserialize, Serialize, Clone, Debug)]
+#[serde(rename_all = "kebab-case")]
+pub struct TomlWorkspaceDependency {
```

Comment:
Side note: we may support artifact dependencies for workspace.

Reply:
What do you mean by this?

Reply:
Other fields for unstable feature [artifact dependencies](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#artifact-dependencies)

```toml
[workspace.dependencies]
bar = { artifact = "cdylib", version = "1.0", lib = true, target = "wasm32-unknown-unknown" }
```

Reply:
I can add those fields in if needed, but they weren't in there before I made these changes

Reply:
Just filed an issue for that: https://github.com/rust-lang/cargo/issues/11625